containers/ecr-viewer/src/app/services/ecrMetadataService.ts

```diff
@@ -96,35 +101,64 @@ export const evaluateEcrMetadata = (fhirBundle: Bundle): EcrMetadata => {
+    fhirBundle,
+    fhirPathMappings.eICRProcessingStatus,
+  );
+  const fhirEICRProcessingStatusReasonObs = evaluateOne(
     fhirBundle,
-    fhirPathMappings.eRSDwarnings,
+    fhirPathMappings.eICRProcessingStatusReason,
   );
-  const eRSDTextList: ERSDWarning[] = [];
 
-  for (const warning of fhirERSDWarnings) {
-    if (warning.code === "RRVS34") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an malformed eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using could not be determined. The trigger codes may be out date. Please have your EHR administrators update the version format for complete eCR functioning.",
-      });
-    } else if (warning.code === "RRVS29") {
-      eRSDTextList.push({
-        warning:
-          "Sending organization is using an outdated eRSD (RCTC) version",
-        versionUsed: "2020-06-23",
-        expectedVersion:
-          "Sending organization should be using one of the following: 2023-10-06, 1.2.2.0, 3.x.x.x.",
-        suggestedSolution:
-          "The trigger code version your organization is using is out-of-date. Please have your EHR administration install the current version for complete eCR functioning.",
-      });
-    }
-  }
+  const eRSDTextList: ERSDWarning[] =
+    fhirEICRProcessingStatus &&
+    fhirEICRProcessingStatus !== "RRVS19" &&
+    fhirEICRProcessingStatusReasonObs
+      ? (() => {
+          const coding =
+            fhirEICRProcessingStatusReasonObs.valueCodeableConcept?.coding?.[0];
+          const warningCode = coding?.code;
+          const warningName =
+            coding?.display || eicrProcessingReasonMap[warningCode ?? ""] || "";
+
+          let versionUsed: string | undefined;
+          let versionExpected: string | undefined;
+
+          (fhirEICRProcessingStatusReasonObs.component ?? []).forEach(
+            (component) => {
+              const detailVal = component.valueString;
+              const detailCode = component.code?.coding?.[0]?.code;
+              const detailDisplay = component.code?.coding?.[0]?.display;
+
+              if (!detailCode) return;
+
+              if (
+                ReasonDetailMap[warningCode as keyof typeof ReasonDetailMap] ===
+                detailCode
+              ) {
+                versionUsed = detailDisplay
+                  ? `${detailDisplay}: ${detailVal}`
+                  : detailVal;
+              } else if (detailCode === "RRVS33") {
+                versionExpected = detailVal;
+              }
+            },
+          );
+
+          return warningName || versionUsed || versionExpected || warningCode
+            ? [
+                {
+                  warning: warningName,
+                  versionUsed: versionUsed ?? "No data",
```

Comment:
should these be the stylized `No Data` to match? if not, should they instead be undefined/optional?